akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -50,6 +50,10 @@ private[akka] object Shard {
    */
   @SerialVersionUID(1L) final case class EntityStopped(entityId: EntityId) extends StateChange
 
+  @SerialVersionUID(1L) final case object GetCurrentShardState
```

Comment:
I didn't make it public since you would anyways go through the `ShardRegion` with the messages `GetShardRegionState`/`CurrentShardRegionState`, considered it an internal impl detail. Motivation was partly that there is no public access to the actual `Shard` actor anywhere else.

You think it should be a public API?